File: android/src/main/java/com/github/simonerm/Job.java
package com.github.simonerm;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

import java.sql.Date;

@Entity(tableName = "job")
public class Job {
    @NonNull
    @PrimaryKey
    private String id;

    @ColumnInfo(name = "worker_name")
    private String workerName;

    private int active;

    private String payload;

    private String metaData;

    private int attempts;

    private String created;

    private int success;

    private int timeout;

    private int priority;

    private String failed;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getWorkerName() {
        return workerName;
    }

    public void setWorkerName(String workerName) {
        this.workerName = workerName;
    }

    public int getActive() {
        return active;
    }

    public void setActive(int active) {
        this.active = active;
    }

    public String getPayload() {
        return payload;
    }

    public void setPayload(String payload) {
        this.payload = payload;
    }

    public int getAttempts() {
        return attempts;
    }

    public void setAttempts(int attempts) {
        this.attempts = attempts;
    }

    public String getCreated() {
        return created;
    }

    public void setCreated(String created) {
        this.created = created;
    }

    public int getSuccess() {
        return success;
    }

    public void setSuccess(int success) {
        this.success = success;
    }

    public int getTimeout() {
        return timeout;
    }

    public void setTimeout(int timeout) {
        this.timeout = timeout;
    }

    public String getFailed() {
        return failed;
    }

    public void setFailed(String failed) {
        this.failed = failed;
    }

    public int getPriority() {
        return priority;
    }

    public void setPriority(int priority) {
        this.priority = priority;
    }

    public String getMetaData() {
        return metaData;
    }

    public void setMetaData(String metaData) {
        this.metaData = metaData;
    }
}

File: src/models/Job.ts
/**
 * represents Job model with Generics
 * @typeparam P payload type
 */
export interface Job<P extends object> {
    id: string;
    workerName: string;
    active: Bool;
    payload: P;
    metaData: string;
    attempts: number;
    created: string;
    failed: string;
    timeout: number;
    priority: number;
}
/**
 * represents native Job model
 */
export interface RawJob {
    id: string;
    workerName: string;
    active: Bool;
    payload: string;
    metaData: string;
    attempts: number;
    created: string;
    failed: string;
    timeout: number;
    priority: number;
}
/**
 * used to map booleans to integer since Sqlite doesn't support boolean
 */
export type Bool = TRUE | FALSE;
type FALSE = 0;
type TRUE = 1;
export const FALSE = 0;
export const TRUE = 1;
